### PR TITLE
Another try to fix TestInspectExecID

### PR DESCRIPTION
**Pull request by duglin**
Signed-off-by: Doug Davis dug@us.ibm.com

### CODE REVIEW

**Comment by duglin** [comment]
Ran windows twice and it worked both times.

File: integration-cli/docker_cli_exec_test.go [hunk 1]
```diff
@@ -349,9 +348,16 @@ func (s *DockerSuite) TestInspectExecID(c *check.C) {
...
+	// End the exec by creating the missing file
+	err = exec.Command(dockerBinary, "exec", "-i", id,
+		"sh", "-c", "touch /tmp/execid1").Run()
+
+	if err != nil {
+		c.Fatalf("failed to run the 2nd exec cmd: %q", err)
+	}
+
+	// Give the first exec time to detect the file
+	time.Sleep(2 * time.Second)
```
Comment (cpuguy83) [inline integration-cli/docker_cli_exec_test.go]: Instead can save the cmd for the first exec to a var, then call cmd.Wait() here.

Comment (duglin) [inline integration-cli/docker_cli_exec_test.go]: Sure, let me give that a try....  anything to remove a sleep!!  :-)

**Comment by duglin** [comment]
seems to work... trying windows again...

**Comment by duglin** [comment]
on to try number 3...  so far so good.

**Comment by duglin** [comment]
ping @jhowardmsft what do you think?  Seems to work on windows.

**Comment by icecrime** [comment]
So if I understand correctly, you think there is some Windows specific behavior of the interactive mode that make `read` exist prematurely?

File: integration-cli/docker_cli_exec_test.go [hunk 2]
```diff
@@ -349,8 +348,15 @@ func (s *DockerSuite) TestInspectExecID(c *check.C) {
 		c.Fatalf("failed to get the exec id: %v", err)
 	}
 
-	// End the exec by closing its stdin, and wait for it to end
-	execStdin.Close()
+	// End the exec by creating the missing file
+	err = exec.Command(dockerBinary, "exec", "-i", id,
```
Comment (icecrime) [inline integration-cli/docker_cli_exec_test.go]: You probably don't need `-i` here (not that it makes a huge difference).

**Comment by duglin** [comment]
yes, and this issue https://github.com/docker/docker/issues/15274  was opened to track it.

**Comment by LK4D4** [comment]
I think probably there is same issue with linux if you use remote daemon.

**Comment by jhowardmsft** [comment]
@duglin Sure, I'll give it a whirl locally. But I'm not convinced looking at it. The problem is that cmd.Wait is local to the client and returns when the docker CLI returns. That's not necessarily in sync with what's happening in the container. @icecrime It appears that way, but I have a hunch it might also be that the CLI is remote to the daemon. That scenario doesn't run in Linux CI.

**Comment by duglin** [comment]
removed the "-i" from the cmds per @icecrime's suggestion

**Comment by jhowardmsft** [comment]
@duglin 3 out of 3 local passes so far, but there's something not quite right - I'm consistently seeing an error in the daemon log HTTP Error err=No such exec instance '<ID>' found in daemon statusCode=404

**Comment by duglin** [comment]
for posterity... the error @jhowardmsft referenced is ok and expected since the test validates that we get a 404 when we try to GET an exec that no longer exists.

**Comment by jhowardmsft** [comment]
30 out of 30 local passes. LGTM.

**Comment by LK4D4** [comment]
LGTM

**Comment by LK4D4** [comment]
@duglin Is this closes some other PRs or issues?

**Comment by duglin** [comment]
@LK4D4 I think it might close PR #14697 - @jhowardmsft is that correct?

**Comment by jhowardmsft** [comment]
Yup.

**Comment by icecrime** [comment]
LGTM

**Comment by thaJeztah** [comment]
test was renamed to `TestExecInspectID`  in https://github.com/moby/moby/commit/a9379b4af227f1966b71ee255fc8bcf272ffe43c

**Pull request merged**